File: core/src/main/java/com/highpowerbear/hpbanalytics/model/Statistics.java
package com.highpowerbear.hpbanalytics.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 *
 * Created by robertk on 4/26/2015.
 */
public class Statistics implements Serializable {
    private static final long serialVersionUID = 8464224239795026258L;

    private final int id;
    private final LocalDateTime periodDate;
    private final int numExecs;
    private final int numOpened;
    private final int numClosed;
    private final int numWinners;
    private final int numLosers;
    private final double pctWinners;
    private final BigDecimal bigWinner;
    private final BigDecimal bigLoser;
    private final BigDecimal winnersProfit;
    private final BigDecimal losersLoss;
    private final BigDecimal profitLoss;
    private final BigDecimal profitLossTaxReport;
    private final BigDecimal cumulProfitLoss;

    public Statistics(int id, LocalDateTime periodDate, int numExecs, int numOpened,int numClosed, int numWinners, int numLosers, double pctWinners,
                      BigDecimal bigWinner, BigDecimal bigLoser, BigDecimal winnersProfit, BigDecimal losersLoss, BigDecimal profitLoss, BigDecimal profitLossTaxReport, BigDecimal cumulProfitLoss) {

        this.id = id;
        this.periodDate = periodDate;
        this.numExecs = numExecs;
        this.numOpened = numOpened;
        this.numClosed = numClosed;
        this.numWinners = numWinners;
        this.numLosers = numLosers;
        this.pctWinners = pctWinners;
        this.bigWinner = bigWinner;
        this.bigLoser = bigLoser;
        this.winnersProfit = winnersProfit;
        this.losersLoss = losersLoss;
        this.profitLoss = profitLoss;
        this.profitLossTaxReport = profitLossTaxReport;
        this.cumulProfitLoss = cumulProfitLoss;
    }

    public int getId() {
        return id;
    }

    public LocalDateTime getPeriodDate() {
        return periodDate;
    }

    public int getNumExecs() {
        return numExecs;
    }

    public int getNumOpened() {
        return numOpened;
    }

    public int getNumClosed() {
        return numClosed;
    }

    public int getNumWinners() {
        return numWinners;
    }

    public int getNumLosers() {
        return numLosers;
    }

    public double getPctWinners() {
        return pctWinners;
    }

    public BigDecimal getBigWinner() {
        return bigWinner;
    }

    public BigDecimal getBigLoser() {
        return bigLoser;
    }

    public BigDecimal getWinnersProfit() {
        return winnersProfit;
    }

    public BigDecimal getLosersLoss() {
        return losersLoss;
    }

    public BigDecimal getProfitLoss() {
        return profitLoss;
    }

    public BigDecimal getProfitLossTaxReport() {
        return profitLossTaxReport;
    }

    public BigDecimal getCumulProfitLoss() {
        return cumulProfitLoss;
    }

    @Override
    public String toString() {
        return "Statistics{" +
                "id=" + id +
                ", periodDate=" + periodDate +
                ", numExecs=" + numExecs +
                ", numOpened=" + numOpened +
                ", numClosed=" + numClosed +
                ", numWinners=" + numWinners +
                ", numLosers=" + numLosers +
                ", pctWinners=" + pctWinners +
                ", bigWinner=" + bigWinner +
                ", bigLoser=" + bigLoser +
                ", winnersProfit=" + winnersProfit +
                ", losersLoss=" + losersLoss +
                ", profitLoss=" + profitLoss +
                ", profitLossTaxReport=" + profitLossTaxReport +
                ", cumulProfitLoss=" + cumulProfitLoss +
                '}';
    }
}

File: core/src/main/java/com/highpowerbear/hpbanalytics/model/Statistics.kt
package com.highpowerbear.hpbanalytics.model

import java.io.Serializable
import java.time.LocalDateTime
import java.math.BigDecimal

/**
 *
 * Created by robertk on 4/26/2015.
 */
class Statistics : Serializable {
    var id = 0
    var periodDate: LocalDateTime? = null
    var numExecs = 0
    var numOpened = 0
    var numClosed = 0
    var numWinners = 0
    var numLosers = 0
    var pctWinners = 0.0
    var bigWinner: BigDecimal = BigDecimal.ZERO
    var bigLoser: BigDecimal = BigDecimal.ZERO
    var winnersProfit: BigDecimal = BigDecimal.ZERO
    var losersLoss: BigDecimal = BigDecimal.ZERO
    var timeValueBought: BigDecimal = BigDecimal.ZERO
    var timeValueSold: BigDecimal = BigDecimal.ZERO
    var timeValueSum: BigDecimal = BigDecimal.ZERO
    var profitLoss: BigDecimal = BigDecimal.ZERO
    var profitLossTaxReport: BigDecimal = BigDecimal.ZERO
    var cumulProfitLoss: BigDecimal = BigDecimal.ZERO

    override fun toString(): String {
        return "Statistics{" +
                "id=" + id +
                ", periodDate=" + periodDate +
                ", numExecs=" + numExecs +
                ", numOpened=" + numOpened +
                ", numClosed=" + numClosed +
                ", numWinners=" + numWinners +
                ", numLosers=" + numLosers +
                ", pctWinners=" + pctWinners +
                ", bigWinner=" + bigWinner +
                ", bigLoser=" + bigLoser +
                ", winnersProfit=" + winnersProfit +
                ", losersLoss=" + losersLoss +
                ", timeValueBought=" + timeValueBought +
                ", timeValueSold=" + timeValueSold +
                ", timeValueSum=" + timeValueSum +
                ", profitLoss=" + profitLoss +
                ", profitLossTaxReport=" + profitLossTaxReport +
                ", cumulProfitLoss=" + cumulProfitLoss +
                '}'
    }

    companion object {
        private const val serialVersionUID = 8464224239795026258L
    }
}